### PR TITLE
Plugins: Remove registry dependency from process manager

### PR DESCRIPTION
**Pull request by wbrowne**
**What is this feature?**

Simplify the plugin process manager interface by passing everything it needs via its methods, which removes the need for a dependency on the plugin registry.

**Why do we need this feature?**

Currently there's quite a dependency placed on the internal plugin registry between a number of plugin management components. Removing the responsibility of fetching the plugin from within the process manager, means the responsibility can be elsewhere, which removes the dependency on the registry.

**Who is this feature for?**

Plugins Platform

**Special notes for your reviewer:**

I also decided to renamed the process manager interface from `Service` to `Manager` (which I believe was standardized somewhere at some point, but can't find)

Please check that:
- [X] It works as expected from a user's perspective.
- [X] If this is a pre-GA feature, it is behind a feature toggle.
- [X] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/plugins/manager/process/process.go [hunk 1]
```diff
@@ -123,7 +67,7 @@ func restartKilledProcess(ctx context.Context, p *plugins.Plugin) error {
 			if err := ctx.Err(); err != nil && !errors.Is(err, context.Canceled) {
 				return err
 			}
-			return nil
+			return p.Stop(ctx)
```
Comment (marefr) [inline pkg/plugins/manager/process/process.go:70]: Could there be an error and the plugin continues to run/are not stopped? Or will `<Store service>.shutdown(ctx)` be called after this returns when plugin already stopped?

File: pkg/plugins/manager/store/store.go [hunk 2]
```diff
@@ -27,6 +29,12 @@ func ProvideService(pluginRegistry registry.Service, pluginSources sources.Regis
 	return New(pluginRegistry), nil
 }
 
+func (s *Service) Run(ctx context.Context) error {
+	<-ctx.Done()
+	s.shutdown(ctx)
```
Comment (marefr) [inline pkg/plugins/manager/store/store.go:34]: Hmm should the store be responsible for shutting down plugins?

**Review by marefr — COMMENTED** [review]
Some comments. In general looks good, but a bit hard to follow who calls whom and what hence my somewhat confusing comments :smile:

File: pkg/plugins/manager/process/process.go (hunk 1 above)
Comment (wbrowne) [reply pkg/plugins/manager/process/process.go:70]: >Could there be an error and the plugin continues to run/are not stopped?

This will only happen when `errors.Is(err, context.Canceled)`, which I _think_ makes sense to try stop the plugin here. If it fails here, then `Store.shutdown` will be called on Grafana shutdown. 

>will <Store service>.shutdown(ctx) be called after this returns when plugin already stopped?

It could, but calling Stop() multiple times is idempotent.

File: pkg/plugins/manager/store/store.go (hunk 2 above)
Comment (wbrowne) [reply pkg/plugins/manager/store/store.go:34]: It's responsible for loading them so for the current state I would say yes 😅

Comment (marefr) [reply pkg/plugins/manager/store/store.go:34]: Yes, but using the loader :) Maybe it should call loader.Unload then? :shrug: No big deal :)

**Review by marefr — APPROVED** [review]
LGTM

File: pkg/plugins/manager/store/store.go (hunk 2 above)
Comment (wbrowne) [reply pkg/plugins/manager/store/store.go:34]: It's a good point you make. You see it's a bit confusing right now because `Unload` will actually remove from disk (it is only called from the uninstall flow), whereas `Load` works off what is already on disk / available - so they're actually mixing things a bit. This ties in with another idea I have so I'll think on this a bit more before merging!

Comment (marefr) [reply pkg/plugins/manager/store/store.go:34]: Don't want to block. Feel free to merge and continue in a new PR
